### DISCUSSION TITLE
MySQL datasource allows to customize connection string

### DISCUSSION THREAD

**Discussion by silencev** [grafana/grafana#37010] [Data Sources]
**What would you like to be added**:
In source code [here](https://github.com/grafana/grafana/blob/b9cc476eb7aa57ed6714654e6036c73c9d105919/pkg/services/sqlstore/sqlstore.go#L229) and [here] I found the connection string parameters are hardcoded.

In MySQL Go driver, it supports following parameters
https://github.com/go-sql-driver/mysql#allowcleartextpasswords

**Why is this needed**:
In my use case, I need to add **allowCleartextPasswords=true** in connection string.

It would be convenient for users to has its own parameters.

**Comment by alexanderilyin** [comment]
This is required when using MySQL 8.0 as Data Source.

**Comment by soufianez75** [comment]
Hello I have the same error it's possible to tell me how you have corrected the data source ?

**Comment by marefr** [comment]
@silencev @alexanderilyin are you referring to using MySQL as Grafana config database or querying MySQL using a Grafana data source for plotting data in a dashboard? Or both? Based on referenced code I'm not sure.

**Reply by Neo-J** [reply]
I think they were taking about using MySQL as datasouce to plot data in dashboard. Correct me if I'm wrong, related code might be [here](https://github.com/grafana/grafana/blob/23956557d8c6a119b7de5be5c42024e29634d002/pkg/tsdb/mysql/mysql.go#L78). From the debug log in console, I can see logger=tsdb.mysql when connecting test triggered from frontend.

This is a very useful feature when MySQL server instance version is later than 8.

**Comment by alexanderilyin** [comment]
In my case I was adding a new MySQL 8 data source for plotting some data on graphs. There was a problem with no default timezone set in MySQL DB and due to some other restrictions, my queries from the panel were failing because of timezone problem. I don't recall how I worked it around but probably found a hacky way. If there was a way to add additional params to the connection string it would be easy to deal with but it's not the option due to how its hardcoded in sources when I checked.

**Comment by junneyang** [comment]
+1 
need connection string parameters:
allowCleartextPasswords=1

**Reply by nicolaerosia** [reply]
this was not fixed

**Reply by SpinoPi** [reply]
What a pity!

**Comment by zoltanbedi** [comment]
This will be available in grafana 10.1.0!
